### PR TITLE
Documentation for becoming a committer, reorganize contributing pages

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

No issue for this one, but it came up recently while adding a new committer that having a page describing the benefits of becoming a committer might be helpful!

## Description

This PR makes a few changes to the documentation around contribution:
- Adds a "current maintainers" page which lists the current active set of maintainers
- Moves many of the contribution-related meta pages into a `contribution` subfolder in the documentation (redirects maintained)
- Adds a page for what a committer is and what the benefits are, based on Apache Airflow's documentation

## Testing Instructions

Check the rendered pages and make sure they make sense!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
Nice improvements!

File: documentation/meta/contribution/index.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Contributing to Openverse
+
+We have the basic outlines of contributing to the project defined in our
+[welcome page](/general/contributing.md). The blow documents provide more
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply documentation/meta/contribution/index.md]: Thanks Olga!

File: documentation/general/contributing.md [hunk 2]
```diff
@@ -280,8 +281,8 @@ Issues may become invalid for many reasons. The most common are:
 In all of these cases it is important to ping the author of the issue and the
 people discussing it. For clarification on whether an issue is still desired, it
 is especially helpful to ping the
-[communication alias](/meta/communication_aliases.md) relevant for the part of
-the stack in question. Please heed
+[communication alias](/meta/contribution/communication_aliases.md) relevant for
+the part of the stack in question. Please heed
```
Comment (stacimc) [inline documentation/general/contributing.md:285]:
```suggestion
the part of the stack in question.
```

**Review by stacimc — APPROVED** [review]
LGTM!

**Pull request merged**